## Summary

### tl;dr

Notes:
* Valkey is a Linux Foundation Project and was forked from Redis on 2024-03-28
* Analysis of contributions to one repo: https://github.com/valkey-io/valkey

There are 18 people who previously contributed to Redis and are now contributing to Valkey. There are a diverse set of contributors from a variety of companies with Amazon having the most contributors. 

Note: The Google data is a bit misleading. While Google employees have made a large percentage of additions and deletions, most of those changes (~28K additions and ~30K deletions) can be attributed to 2 PRs ([#323](https://github.com/valkey-io/valkey/pull/323) and [#591](https://github.com/valkey-io/valkey/pull/591)), which were related to formatting changes.

### 1 Year After 2024-03-28 Fork (until 2025-03-28):
|People|Organization|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
|29|Amazon|335|42314 (25.13%)|15759 (15.99%)|
|4|Alibaba|20|2049 (1.22%)|419 (0.43%)|
|2|Google|46|44524 (26.44%)|40971 (41.58%)|
|2|Ericsson|110|22282 (13.23%)|8173 (8.30%)|
|2|Samsung|5|48 (0.03%)|48 (0.05%)|
|2|ByteDance|22|6903 (4.10%)|2456 (2.49%)|
|1|Percona|14|4591 (2.73%)|2595 (2.63%)|
|1|LINE Corporation|6|130 (0.08%)|100 (0.10%)|
|4|Huawei|97|3702 (2.20%)|3191 (3.24%)|
|2|Intel|22|775 (0.46%)|524 (0.53%)|
|1|@gnet-io|8|104 (0.06%)|73 (0.07%)|
|1|@Aliyun|6|107 (0.06%)|67 (0.07%)|
|1|Tencent Cloud|200|9337 (5.54%)|4513 (4.58%)|

Companies with >= 5 commits: 
* People: 52 37.14% of people employed at 13 companies
* Commits: 891
* Additions: 136866 81.27% of total additions
* Deletions: 78889 80.07% of total deletions

### 6 Months After 2024-03-28 Fork (until 2024-09-28):

|People|Organization|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
|2|Alibaba|12|585 (0.59%)|93 (0.13%)|
|13|Amazon|149|18232 (18.31%)|6288 (8.66%)|
|2|Samsung|5|48 (0.05%)|48 (0.07%)|
|2|Ericsson|45|5867 (5.89%)|1954 (2.69%)|
|4|Huawei|76|3561 (3.58%)|3016 (4.16%)|
|1|Tencent Cloud|92|4859 (4.88%)|2429 (3.35%)|
|2|ByteDance|7|3952 (3.97%)|572 (0.79%)|
|2|Google|39|40698 (40.86%)|38643 (53.24%)|
|1|Intel|12|632 (0.63%)|464 (0.64%)|
|1|@gnet-io|8|104 (0.10%)|73 (0.10%)|

Companies with >= 5 commits: 
* People: 30 33.71% of people employed at 10 companies
* Commits: 445
* Additions: 78538 78.86% of total additions
* Deletions: 53580 73.82% of total deletions


# 1 Year after Relicense (2024-03-28 - 2025-03-28)

In [1]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/chaoss/wg-data-science/blob/main/dataset/license-changes/fork-case-study/commits_people.py

people_pickle = '../data-files/valkey_people_2024-03-28T00:00:00.000+00:002025-03-28T00:00:00.000+00:00.pkl'

with open(people_pickle, 'rb') as f:
    person_dict = pickle.load(f)
    
len(person_dict)

140

In [4]:
people = len(person_dict)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict.items():
    # Normalize company names and use emails to derive some affiliations
    print(key,value)
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict[key]['company'] = 'Amazon'
            if 'intel.com' in email:
                person_dict[key]['company'] = 'Intel'
    elif 'huawei' in value['company'].lower():
        person_dict[key]['company'] = 'Huawei'
    elif 'alibaba' in value['company'].lower():
        person_dict[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict[key]['company'] = 'Amazon'
    elif 'ericsson' in value['company'].lower():
        person_dict[key]['company'] = 'Ericsson'
    elif 'samsung' in value['company'].lower():
        person_dict[key]['company'] = 'Samsung'
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

roshkhatri {'name': 'Roshan Khatri', 'email': ['117414976+roshkhatri@users.noreply.github.com'], 'company': None, 'additions': 849, 'deletions': 147, 'commits': 14}
pshankinclarke {'name': None, 'email': ['38367844+pshankinclarke@users.noreply.github.com'], 'company': None, 'additions': 3, 'deletions': 1, 'commits': 2}
DarrenJiang13 {'name': None, 'email': ['yjjiang1996@163.com'], 'company': 'Shanghai Jiaotong University', 'additions': 2, 'deletions': 0, 'commits': 1}
knggk {'name': 'Guillaume Koenig', 'email': ['106696198+knggk@users.noreply.github.com'], 'company': 'Amazon', 'additions': 1129, 'deletions': 36, 'commits': 2}
Nugine {'name': None, 'email': ['nugine@foxmail.com'], 'company': None, 'additions': 345, 'deletions': 11, 'commits': 1}
danish-mehmood {'name': None, 'email': ['35922417+danish-mehmood@users.noreply.github.com'], 'company': None, 'additions': 1, 'deletions': 1, 'commits': 1}
azuredream {'name': 'zixuan zhao', 'email': ['zhaozixuan67@gmail.com'], 'company': None, 

In [6]:
for key,value in person_dict.items():
    if (value['commits'] >= 5) and (value['company'] == None):
        print(key,value)
        
# Note: Unaffiliated after Aug 2023: https://www.linkedin.com/in/9bany/

roshkhatri {'name': 'Roshan Khatri', 'email': ['117414976+roshkhatri@users.noreply.github.com'], 'company': None, 'additions': 849, 'deletions': 147, 'commits': 14}
bentotten {'name': None, 'email': ['59932872+bentotten@users.noreply.github.com'], 'company': None, 'additions': 25, 'deletions': 6, 'commits': 8}
kyle-yh-kim {'name': 'Kyle Kim (kimkyle@)', 'email': ['105247741+kyle-yh-kim@users.noreply.github.com'], 'company': None, 'additions': 1797, 'deletions': 127, 'commits': 5}
vitahlin {'name': 'Vitah Lin', 'email': ['vitahlin@gmail.com', 'linw1225@gmail.com'], 'company': None, 'additions': 144, 'deletions': 57, 'commits': 9}
parthpatel {'name': 'Parth', 'email': ['661497+parthpatel@users.noreply.github.com'], 'company': None, 'additions': 4298, 'deletions': 103, 'commits': 5}
xbasel {'name': None, 'email': ['103044017+xbasel@users.noreply.github.com'], 'company': None, 'additions': 159, 'deletions': 66, 'commits': 6}
eliblight {'name': 'Eran Liberty', 'email': ['eran.liberty@gmail.

In [8]:
# Manual Fixes
person_dict['roshkhatri']['company'] = 'Amazon' # https://www.linkedin.com/in/roshan-vivek-khatri-a403a8193/
person_dict['bentotten']['company'] = 'Amazon' # https://www.linkedin.com/in/ben-totten/
person_dict['uriyage']['company'] = 'Amazon' # https://valkey.io/blog/unlock-one-million-rps/
person_dict['eliblight']['company'] = 'Amazon' # https://www.linkedin.com/in/eran-liberty-b7317b4/
person_dict['kyle-yh-kim']['company'] = 'Amazon' # https://www.linkedin.com/in/kyle-yong-kim/
person_dict['CharlesChen888']['company'] = 'Alibaba' # confirmed via email by Madelyn Olson 2024-10-21

In [9]:
import collections

org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

orgs = 0

org_dict = collections.defaultdict(dict)

for key,value in person_dict.items():
    try:
        org_dict[value['company']]['commits'] = org_dict[value['company']]['commits'] + value['commits']
        org_dict[value['company']]['additions'] = org_dict[value['company']]['additions'] + value['additions']
        org_dict[value['company']]['deletions'] = org_dict[value['company']]['deletions'] + value['deletions']
        org_dict[value['company']]['people'].append(key)
    except:
        org_dict[value['company']]['commits'] = value['commits']
        org_dict[value['company']]['additions'] = value['additions']
        org_dict[value['company']]['deletions'] = value['deletions']
        org_dict[value['company']]['people'] = [key]

        
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Organization|Commits|Additions|Deletions|')
print('|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
    if value['commits'] >= 5:
        if key != None:
            org_people = org_people + len(value['people'])
            org_commits = org_commits + value['commits']
            org_additions = org_additions + value['additions']
            org_deletions = org_deletions + value['deletions']
            orgs+=1
            add_perc = format(value['additions']/additions, ".2%")
            del_perc = format(value['deletions']/deletions, ".2%")
            print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],' (',add_perc,')|', value['deletions'],' (',del_perc,')|',sep='')

print("\nCompanies with >= 5 commits:", "\n* People:", org_people, format(org_people/people, ".2%"), "of people employed at", orgs, "companies")
print("* Commits:", org_commits)
print("* Additions:", org_additions, format(org_additions/additions, ".2%"), "of total additions")
print("* Deletions:", org_deletions, format(org_deletions/deletions, ".2%"), "of total deletions")

|People|Organization|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
|29|Amazon|335|42314 (25.13%)|15759 (15.99%)|
|4|Alibaba|20|2049 (1.22%)|419 (0.43%)|
|2|Google|46|44524 (26.44%)|40971 (41.58%)|
|2|Ericsson|110|22282 (13.23%)|8173 (8.30%)|
|2|Samsung|5|48 (0.03%)|48 (0.05%)|
|2|ByteDance|22|6903 (4.10%)|2456 (2.49%)|
|1|Percona|14|4591 (2.73%)|2595 (2.63%)|
|1|LINE Corporation|6|130 (0.08%)|100 (0.10%)|
|4|Huawei|97|3702 (2.20%)|3191 (3.24%)|
|2|Intel|22|775 (0.46%)|524 (0.53%)|
|1|@gnet-io|8|104 (0.06%)|73 (0.07%)|
|1|@Aliyun|6|107 (0.06%)|67 (0.07%)|
|1|Tencent Cloud|200|9337 (5.54%)|4513 (4.58%)|

Companies with >= 5 commits: 
* People: 52 37.14% of people employed at 13 companies
* Commits: 891
* Additions: 136866 81.27% of total additions
* Deletions: 78889 80.07% of total deletions


In [10]:
i = 0
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Organization|Commits|Additions|Addition%|Deletions|Deletion%|')
print('|:---|:---|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
           
    add_perc = format(value['additions']/additions, ".2%")
    del_perc = format(value['deletions']/deletions, ".2%")
    print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],'|',add_perc,'|', value['deletions'],'|',del_perc,'|')

    i+=1

print("Number of companies:", i)

|People|Organization|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 29 | Amazon | 335 | 42314 | 25.13% | 15759 | 15.99% |
| 53 | None | 132 | 15838 | 9.40% | 4850 | 4.92% |
| 1 | Shanghai Jiaotong University | 1 | 2 | 0.00% | 0 | 0.00% |
| 4 | Alibaba | 20 | 2049 | 1.22% | 419 | 0.43% |
| 1 | iandouglas.com | 1 | 2 | 0.00% | 2 | 0.00% |
| 1 | @JaneaSystems  | 1 | 31 | 0.02% | 6 | 0.01% |
| 2 | Google | 46 | 44524 | 26.44% | 40971 | 41.58% |
| 2 | Ericsson | 110 | 22282 | 13.23% | 8173 | 8.30% |
| 1 | Oracle | 4 | 10948 | 6.50% | 10948 | 11.11% |
| 2 | Samsung | 5 | 48 | 0.03% | 48 | 0.05% |
| 1 | @valkey-io | 1 | 0 | 0.00% | 6 | 0.01% |
| 1 | Amadeus | 1 | 15 | 0.01% | 6 | 0.01% |
| 1 | Humana | 1 | 1087 | 0.65% | 984 | 1.00% |
| 2 | ByteDance | 22 | 6903 | 4.10% | 2456 | 2.49% |
| 1 | Percona | 14 | 4591 | 2.73% | 2595 | 2.63% |
| 1 | LINE Corporation | 6 | 130 | 0.08% | 100 | 0.10% |
| 1 | @percona | 1 | 2 | 0.00% | 2 | 0.00% |
| 2 | @aiven | 

# 6 Months after Relicense (2024-03-28 - 2024-09-28)

In [1]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/chaoss/wg-data-science/blob/main/dataset/license-changes/fork-case-study/commits_people.py

people_pickle = '../data-files/valkey_people_2024-03-28T00:00:00.000+00:002024-09-28T00:00:00.000+00:00.pkl'

with open(people_pickle, 'rb') as f:
    person_dict = pickle.load(f)
    
len(person_dict)

89

In [2]:
people = len(person_dict)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict.items():
    # Normalize company names and use emails to derive some affiliations
    print(key,value)
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict[key]['company'] = 'Amazon'
    elif 'huawei' in value['company'].lower():
        person_dict[key]['company'] = 'Huawei'
    elif 'alibaba' in value['company'].lower():
        person_dict[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict[key]['company'] = 'Amazon'
    elif 'ericsson' in value['company'].lower():
        person_dict[key]['company'] = 'Ericsson'
    elif 'samsung' in value['company'].lower():
        person_dict[key]['company'] = 'Samsung'
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

Yury-Fridlyand {'additions': 41, 'email': ['yury.fridlyand@improving.com'], 'company': None, 'deletions': 3, 'commits': 1, 'name': None}
soloestoy {'additions': 415, 'email': ['zhaozhao.zz@alibaba-inc.com'], 'company': 'Alibaba', 'deletions': 71, 'commits': 8, 'name': 'zhaozhao.zz'}
shogo82148 {'additions': 41, 'email': ['shogo82148@gmail.com'], 'company': '@fuller-inc ', 'deletions': 38, 'commits': 1, 'name': 'ICHINOSE Shogo'}
madolson {'additions': 8642, 'email': ['madelyneolson@gmail.com', '34459052+madolson@users.noreply.github.com'], 'company': 'Amazon Web Services', 'deletions': 3829, 'commits': 76, 'name': 'Madelyn Olson'}
hallmason17 {'additions': 1087, 'email': ['55302632+hallmason17@users.noreply.github.com'], 'company': 'Humana', 'deletions': 984, 'commits': 1, 'name': 'Mason Hall'}
arthurkiller {'additions': 25, 'email': ['arthurkiller@users.noreply.github.com'], 'company': None, 'deletions': 4, 'commits': 3, 'name': 'Arthur Lee'}
mapleFU {'additions': 28, 'email': ['maplew

In [3]:
for key,value in person_dict.items():
    if (value['commits'] >= 5) and (value['company'] == None):
        print(key,value)
        
# Note: Unaffiliated after Aug 2023: https://www.linkedin.com/in/9bany/

bentotten {'additions': 21, 'email': ['59932872+bentotten@users.noreply.github.com'], 'company': None, 'deletions': 6, 'commits': 6, 'name': None}
9bany {'additions': 619, 'email': ['53461381+0del@users.noreply.github.com'], 'company': None, 'deletions': 616, 'commits': 26, 'name': 'Bany'}
roshkhatri {'additions': 767, 'email': ['117414976+roshkhatri@users.noreply.github.com'], 'company': None, 'deletions': 113, 'commits': 7, 'name': 'Roshan Khatri'}
vitahlin {'additions': 144, 'email': ['vitahlin@gmail.com', 'linw1225@gmail.com'], 'company': None, 'deletions': 57, 'commits': 9, 'name': 'Vitah Lin'}
kyle-yh-kim {'additions': 1797, 'email': ['105247741+kyle-yh-kim@users.noreply.github.com'], 'company': None, 'deletions': 127, 'commits': 5, 'name': 'Kyle Kim (kimkyle@)'}
uriyage {'additions': 2889, 'email': ['78144248+uriyage@users.noreply.github.com'], 'company': None, 'deletions': 1124, 'commits': 11, 'name': None}
eliblight {'additions': 14, 'email': ['eran.liberty@gmail.com'], 'compa

In [4]:
# Manual Fixes
person_dict['roshkhatri']['company'] = 'Amazon' # https://www.linkedin.com/in/roshan-vivek-khatri-a403a8193/
person_dict['uriyage']['company'] = 'Amazon' # https://valkey.io/blog/unlock-one-million-rps/
person_dict['eliblight']['company'] = 'Amazon' # https://www.linkedin.com/in/eran-liberty-b7317b4/
person_dict['stockholmux']['company'] = 'Amazon' # Using amazon.com email
person_dict['kyle-yh-kim']['company'] = 'Amazon' # https://www.linkedin.com/in/kyle-yong-kim/
person_dict['CharlesChen888']['company'] = 'Alibaba' # confirmed via email by Madelyn Olson 2024-10-21

In [5]:
import collections

org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

orgs = 0

org_dict = collections.defaultdict(dict)

for key,value in person_dict.items():
    try:
        org_dict[value['company']]['commits'] = org_dict[value['company']]['commits'] + value['commits']
        org_dict[value['company']]['additions'] = org_dict[value['company']]['additions'] + value['additions']
        org_dict[value['company']]['deletions'] = org_dict[value['company']]['deletions'] + value['deletions']
        org_dict[value['company']]['people'].append(key)
    except:
        org_dict[value['company']]['commits'] = value['commits']
        org_dict[value['company']]['additions'] = value['additions']
        org_dict[value['company']]['deletions'] = value['deletions']
        org_dict[value['company']]['people'] = [key]

        
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Organization|Commits|Additions|Deletions|')
print('|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
    if value['commits'] >= 5:
        if key != None:
            org_people = org_people + len(value['people'])
            org_commits = org_commits + value['commits']
            org_additions = org_additions + value['additions']
            org_deletions = org_deletions + value['deletions']
            orgs+=1
            add_perc = format(value['additions']/additions, ".2%")
            del_perc = format(value['deletions']/deletions, ".2%")
            print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],' (',add_perc,')|', value['deletions'],' (',del_perc,')|',sep='')

print("\nCompanies with >= 5 commits:", "\n* People:", org_people, format(org_people/people, ".2%"), "of people employed at", orgs, "companies")
print("* Commits:", org_commits)
print("* Additions:", org_additions, format(org_additions/additions, ".2%"), "of total additions")
print("* Deletions:", org_deletions, format(org_deletions/deletions, ".2%"), "of total deletions")

|People|Organization|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
|2|Alibaba|12|585 (0.59%)|93 (0.13%)|
|13|Amazon|149|18232 (18.31%)|6288 (8.66%)|
|2|Samsung|5|48 (0.05%)|48 (0.07%)|
|2|Ericsson|45|5867 (5.89%)|1954 (2.69%)|
|4|Huawei|76|3561 (3.58%)|3016 (4.16%)|
|1|Tencent Cloud|92|4859 (4.88%)|2429 (3.35%)|
|2|ByteDance|7|3952 (3.97%)|572 (0.79%)|
|2|Google|39|40698 (40.86%)|38643 (53.24%)|
|1|Intel|12|632 (0.63%)|464 (0.64%)|
|1|@gnet-io|8|104 (0.10%)|73 (0.10%)|

Companies with >= 5 commits: 
* People: 30 33.71% of people employed at 10 companies
* Commits: 445
* Additions: 78538 78.86% of total additions
* Deletions: 53580 73.82% of total deletions


In [42]:
i = 0
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Organization|Commits|Additions|Addition%|Deletions|Deletion%|')
print('|:---|:---|:---|:---|:---|:---|:---|')

for key,value in org_dict.items():
           
    add_perc = format(value['additions']/additions, ".2%")
    del_perc = format(value['deletions']/deletions, ".2%")
    print('|', len(value['people']),'|', key, '|', value['commits'],'|', value['additions'],'|',add_perc,'|', value['deletions'],'|',del_perc,'|')

    i+=1

print("Number of companies:", i)

|People|Organization|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 36 | None | 95 | 4647 | 4.67% | 2749 | 3.79% |
| 2 | Alibaba | 12 | 585 | 0.59% | 93 | 0.13% |
| 1 | @fuller-inc  | 1 | 41 | 0.04% | 38 | 0.05% |
| 13 | Amazon | 149 | 18232 | 18.31% | 6288 | 8.66% |
| 1 | Humana | 1 | 1087 | 1.09% | 984 | 1.36% |
| 2 | @aiven | 4 | 2125 | 2.13% | 2230 | 3.07% |
| 1 | Soochow University | 1 | 13 | 0.01% | 6 | 0.01% |
| 1 | simkus.io | 2 | 38 | 0.04% | 38 | 0.05% |
| 1 | Appier | 1 | 44 | 0.04% | 3 | 0.00% |
| 2 | Samsung | 5 | 48 | 0.05% | 48 | 0.07% |
| 2 | Ericsson | 45 | 5867 | 5.89% | 1954 | 2.69% |
| 4 | Huawei | 76 | 3561 | 3.58% | 3016 | 4.16% |
| 1 | self - deli clerk | 1 | 1 | 0.00% | 1 | 0.00% |
| 1 | @VelocityLimitless | 1 | 5 | 0.01% | 0 | 0.00% |
| 1 | Prunny Technologies | 3 | 1593 | 1.60% | 1585 | 2.18% |
| 1 | Seeking Opportunities | 3 | 36 | 0.04% | 36 | 0.05% |
| 1 | Tencent Cloud | 92 | 4859 | 4.88% | 2429 | 3.35% |
| 1 | Orac

## All Organizational Affiliations

Note: In some cases, these are based on what people have put in their GitHub profile and aren't actual organizations (e.g., "Looking for a job"), but those are part of the long tail and aren't significant contributors.

|People|Company|Commits|Additions|Addition%|Deletions|Deletion%|
|:---|:---|:---|:---|:---|:---|:---|
| 36 | None | 95 | 4647 | 4.67% | 2749 | 3.79% |
| 1 | Alibaba | 8 | 415 | 0.42% | 71 | 0.10% |
| 1 | @fuller-inc  | 1 | 41 | 0.04% | 38 | 0.05% |
| 13 | Amazon | 149 | 18232 | 18.31% | 6288 | 8.66% |
| 1 | Humana | 1 | 1087 | 1.09% | 984 | 1.36% |
| 2 | @aiven | 4 | 2125 | 2.13% | 2230 | 3.07% |
| 1 | Soochow University | 1 | 13 | 0.01% | 6 | 0.01% |
| 1 | simkus.io | 2 | 38 | 0.04% | 38 | 0.05% |
| 1 | Appier | 1 | 44 | 0.04% | 3 | 0.00% |
| 2 | Samsung | 5 | 48 | 0.05% | 48 | 0.07% |
| 2 | Ericsson | 45 | 5867 | 5.89% | 1954 | 2.69% |
| 4 | Huawei | 76 | 3561 | 3.58% | 3016 | 4.16% |
| 1 | self - deli clerk | 1 | 1 | 0.00% | 1 | 0.00% |
| 1 | @VelocityLimitless | 1 | 5 | 0.01% | 0 | 0.00% |
| 1 | Prunny Technologies | 3 | 1593 | 1.60% | 1585 | 2.18% |
| 1 | Seeking Opportunities | 3 | 36 | 0.04% | 36 | 0.05% |
| 1 | Tencent Cloud | 92 | 4859 | 4.88% | 2429 | 3.35% |
| 1 | Oracle | 4 | 10948 | 10.99% | 10948 | 15.08% |
| 1 | LINE | 3 | 122 | 0.12% | 69 | 0.10% |
| 1 | @franklin-ai | 4 | 137 | 0.14% | 96 | 0.13% |
| 1 | Shanghai Jiaotong University | 1 | 2 | 0.00% | 0 | 0.00% |
| 2 | ByteDance | 7 | 3952 | 3.97% | 572 | 0.79% |
| 1 | The Linux Foundation | 1 | 6 | 0.01% | 0 | 0.00% |
| 2 | Google | 39 | 40698 | 40.86% | 38643 | 53.24% |
| 1 | University of Michigan | 1 | 4 | 0.00% | 8 | 0.01% |
| 1 | Percona | 1 | 1 | 0.00% | 1 | 0.00% |
| 1 | @Aliyun | 2 | 14 | 0.01% | 25 | 0.03% |
| 1 | Intel | 12 | 632 | 0.63% | 464 | 0.64% |
| 1 | @gnet-io | 8 | 104 | 0.10% | 73 | 0.10% |
| 1 | @kunai-consulting | 1 | 15 | 0.02% | 0 | 0.00% |
| 1 | HKU | 4 | 170 | 0.17% | 22 | 0.03% |
| 1 | Zhejiang University | 1 | 0 | 0.00% | 2 | 0.00% |
| 1 | iandouglas.com | 1 | 2 | 0.00% | 2 | 0.00% |
| 1 | Redis | 1 | 175 | 0.18% | 185 | 0.25% |

In [43]:
for key,value in person_dict.items():
    try:
        if value['company'] == 'Redis':
            print(key,value)
    except:
        pass

K-Jo {'additions': 175, 'email': ['pieter@redislabs.com'], 'company': 'Redis', 'deletions': 185, 'commits': 1, 'name': 'Pieter Cailliau'}


# List of people who also contributed to Redis before the relicense

In [11]:
redis_pickle_1yr = '../data-files/redis_people_2023-03-20T00:00:00.000+00:002024-03-20T00:00:00.000+00:00.pkl'

with open(redis_pickle_1yr, 'rb') as f:
    redis_dict_1yr = pickle.load(f)

common_contrib = set(redis_dict_1yr) & set(person_dict)
print("Common Contributors:", common_contrib)

Common Contributors: {'knggk', 'bentotten', 'zuiderkwast', 'YaacovHazan', 'hwware', 'soloestoy', 'panjf2000', 'roshkhatri', 'PingXie', 'LiiNen', 'DarrenJiang13', 'lyq2333', 'CharlesChen888', 'hpatro', 'gabiganam', 'srgsanky', 'madolson', 'pizhenwei', 'secwall', 'enjoy-binbin', 'BCathcart', 'yangbodong22011'}


In [12]:
len(common_contrib)

22

In [13]:
for x in common_contrib:
    print(x, person_dict[x]['company'])

knggk Amazon
bentotten Amazon
zuiderkwast Ericsson
YaacovHazan Redis Labs
hwware Huawei
soloestoy Alibaba
panjf2000 @gnet-io
roshkhatri Amazon
PingXie Google
LiiNen LINE
DarrenJiang13 Shanghai Jiaotong University
lyq2333 Alibaba
CharlesChen888 Alibaba
hpatro Amazon
gabiganam Amazon
srgsanky None
madolson Amazon
pizhenwei ByteDance
secwall None
enjoy-binbin Tencent Cloud
BCathcart None
yangbodong22011 Alibaba
